src/poetry/utils/shell.py

```diff
@@ -67,8 +68,20 @@ def get(cls) -> "Shell":
         return cls._shell
 
     def activate(self, env: "VirtualEnv") -> Optional[int]:
+        activate_script = self._get_activate_script()
+        bin_dir = "Scripts" if WINDOWS else "bin"
+        activate_path = env.path / bin_dir / activate_script
+
         if WINDOWS:
-            return env.execute(self.path)
+            if self._name == "powershell":
+                args = ["-NoExit", "-File", str(activate_path)]
```

Comment:
Subjective opinion:  I like launching with `-nologo` to not have to see

> PowerShell 7.2.1
Copyright (c) Microsoft Corporation.

> https://aka.ms/powershell
Type 'help' to get help.

each time.

Reply:
Given that it is cosmetic, I'd rather not add more arguments past the bare minimum, personally.

Reply:
Understood.  It is definitely subjective on my part -- I like the "clean sheet" in front of me after typing `poetry shell`, but typing `cls` isn't a big deal.